esx_service/utils/vmdk_utils.py

```diff
@@ -95,33 +110,47 @@ def list_vmdks(path, volname="", show_snapshots=False):
     show_snapshots - if set to True, all VMDKs (including delta files) will be returned
     """
 
-    # dockvols may not exists on a datastore
+    # dockvols may not exists on a datastore - this is normal.
     if not os.path.exists(path):
         return []
 
-    glob_pattern = "{0}/{1}*.vmdk".format(path, volname)
-    vmdks = [os.path.basename(f) for f in glob.glob(glob_pattern)
-            if vmdk_is_a_descriptor(os.path.join(path, f))]
+    vmdks = [f for f in os.listdir(path) if vmdk_is_a_descriptor(path, f)]
+    if volname:
+        vmdks = [f for f in vmdks if f.startswith(volname)]
 
-    # For hiding snapshots we rely on volume names NOT having a '-' symbol,
-    # so all files with '-' must be delta disks, digest and the likes, and
-    # need to be hidden from further processing.
-    # see vmdk_ops.py:parse_vol_name() which enforces the "no -" rule.
     if not show_snapshots:
-        vmdks = [f for f in vmdks if f.find('-') == -1]
+        expr =  re.compile(SNAP_VMDK_REGEXP)
+        vmdks = [f for f in vmdks if not expr.match(f)]
     
     return vmdks
 
 
-def vmdk_is_a_descriptor(filepath):
+def vmdk_is_a_descriptor(path, file_name):
     """
-    Is the file a vmdk descriptor file?  We assume any file that ends in .vmdk
-    and has a size less than MAX_DESCR_SIZE is a desciptor file.
+    Is the file a vmdk descriptor file?  We assume any file that ends in .vmdk,
+    does not have -delta or -flat or he likes at the end of filename, 
+    and has a size less than MAX_DESCR_SIZE is a descriptor file.
     """
-    if filepath.endswith('.vmdk') and os.stat(filepath).st_size < MAX_DESCR_SIZE:
-       return True
 
-    return False
+    name = file_name.lower()
+
+    # filter out all files with wrong extention
+    if not name.endswith('.vmdk'):
+        return False
+
+    # filter out -delta, -flat, -digest and -ctk VMDK files
+    if re.match(SPECIAL_FILES_REGEXP, name):
+        return False
+
```

Comment:
Can this be,
if not name.endswith(...) or re.match(SPECIAL....):
     return False